units/db_cleanup_general.go

```diff
@@ -0,0 +1,101 @@
+package units
+
+import (
+	"context"
+	"time"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/mongodb/grip/message"
+	"github.com/pkg/errors"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+const (
+	cleanupBatchSize = 100 * 1000
+)
+
+type FilterFunc func(time.Time) map[string]interface{}
+
+type DataCleanupJobBase struct {
+	CollectionName string        `bson:"collection_name" json:"collection_name" yaml:"collection_name"`
+	TTL            time.Duration `bson:"ttl" json:"ttl" yaml:"ttl"`
+	Errors         []error       `bson:"errors" json:"errors" yaml:"errors"`
+
+	env evergreen.Environment
+}
+
+func (b *DataCleanupJobBase) RunWithDeleteFn(ctx context.Context, filterFunc FilterFunc) (message.Fields, []error) {
```

Comment:
It doesn't seem like this method needs to be exported since no other package uses it - can you make this `runWithDeleteFn`?